### PR TITLE
gh-74166: make all_errors keyword-only

### PR DESCRIPTION
**Pull request by iritkatriel**

### CODE REVIEW

File: Lib/socket.py [hunk 1]
```diff
@@ -806,7 +806,7 @@ def getfqdn(name=''):
 _GLOBAL_DEFAULT_TIMEOUT = object()
 
 def create_connection(address, timeout=_GLOBAL_DEFAULT_TIMEOUT,
-                      source_address=None, all_errors=False):
+                      source_address=None, *, all_errors=False):
```
Comment (iritkatriel) [inline Lib/socket.py:809]: Is it weird that all_errors is kw-only and the others are not?

Comment (vstinner) [reply Lib/socket.py:809]: Nope. It's called "legacy" :-)

**Review by vstinner — APPROVED** [review]
LGTM.

**Comment by vstinner** [comment]
Thanks.

Once I wanted to convert most open() parameter to keyword-only. I never recall what are positional parameter after the first two obvious arguments. But my idea got quickly rejected. "The ship has sailed", it's too late to convert positional-or-keyword parameter to keyword-only, at least without a slow deprecation process. But for new parameter, we have a great freedom to decide how to declare them! Keyword only, positional only, etc.

**Pull request merged**